src/lib/qnio/io_ssl.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Network IO library for VxHS QEMU block driver (Veritas Technologies)
+ *
+ * This work is licensed under the terms of the GNU GPL, version 2.  See
+ * the COPYING file in the top-level directory.
+ *
+ * Contributions after 2014-08-15 are licensed under the terms of the
+ * GNU GPL, version 2 or (at your option) any later version.
+ */
+
+#include "defs.h"
+#include "qnio.h"
+
+static int
+read_ssl(struct endpoint *endpoint, void *buf, size_t len)
+{
+    assert(endpoint->ssl != NULL);
+    return (SSL_read(endpoint->ssl, buf, len));
+}
+
+static int
+write_ssl(struct endpoint *endpoint, const void *buf, size_t len)
+{
+    assert(endpoint->ssl != NULL);
+    return (SSL_write(endpoint->ssl, buf, len));
+}
+
+static int
+readv_ssl(struct endpoint *endpoint, struct iovec *vec, int count) 
+{
+    assert(endpoint->ssl != NULL);
+    return (SSL_read(endpoint->ssl, vec[0].iov_base, vec[0].iov_len));
```

Comment:
The core multiplexer in libqnio calls the readv/writev for io_socket for better performance. To keep the interface consistent for the multiplexer I am using the same call for io_ssl. However since there is no native readv/writev for SSL, this implementation defaults to regular SSL read/write. The iovec adjustments for partial reads/writes are handled in nio_core.c (write_to_network and read_from_network) but the interface remains consistent.